pkg/k8shandler/kibana/reconciler.go

```diff
@@ -393,6 +392,40 @@ func updateCurrentDeploymentImages(current *apps.Deployment, desired *apps.Deplo
 	return current
 }
 
+func isDeploymentEnvDifference(current *apps.Deployment, desired *apps.Deployment) bool {
```

Comment:
Can we merge the two functions into one?
```
func updateCurrentDeploymentEnvIfDifferent(current *apps.Deployment, desired *apps.Deployment) bool {

        different := false

	for _, curr := range current.Spec.Template.Spec.Containers {
		for _, des := range desired.Spec.Template.Spec.Containers {
			// Only compare the env of containers with the same name
			if curr.Name == des.Name {
				if !utils.EnvValueEqual(curr.Env, des.Env) {
                                        curr.Env = des.Env
                                        different = true
				}
			}
		}
	}

	return different
}
```

Reply:
updated